modules/storages/lib/api/v3/storages/storage_representer.rb

```diff
@@ -51,28 +50,33 @@ class StorageRepresenter < ::API::Decorators::Single
 
         date_time_property :updated_at
 
-        # A link back to the specific object ("represented")
         self_link
 
         link :type do
           {
-            href: URN_TYPE_NEXTCLOUD,
+            href: "#{::API::V3::URN_PREFIX}storages:Nextcloud",
             title: 'Nextcloud'
           }
         end
 
         link :origin do
-          {
-            href: represented.host
-          }
+          { href: represented.host }
         end
 
-        link :connectionState do
-          # TODO: replace with service to check real connection state
-          {
-            href: URN_CONNECTION_CONNECTED,
-            title: 'Connected'
-          }
+        link :authorizationState do
+          state = ::API::V3::Storages::StorageAuthorizer.authorize represented
+
+          urn = case state
+                when :connected
+                  URN_CONNECTION_CONNECTED
+                when :failed_authorization
+                  URN_CONNECTION_AUTH_FAILED
+                else
+                  URN_CONNECTION_ERROR
+                end
```

Comment:
I probably wasn't specific enough in our talk. I would keep the URN constants within the `StorageAuthorizer`. You might need that again in a different representer.

Reply:
This was extracted due to the fact, that returning the URNs from the authorizer would require string parsing in the representer. To avoid that, and having the issue, that the URNs are "currently" ot used elsewhere, I'd argue, that it is more sensible to just return a symbol and let the representer translate that very symbol to a) an URN and b) a human readable title.
Alternatively we could simply omit the title in the representer.

Reply:
I agree with @Kharonus for now, as there will some work be done in the next tasks on that representer.